### PR TITLE
P argument deprecation

### LINKED ISSUE #5225: Deprecate the '-p' argument in the 'download' and 'copy' cli
As it is in `conan get...` and soon in `conan upload...`(#5224), it makes sense to deprecate the `-p` argument for `conan download...` and `conan copy...` too. 

----
It still appears in the `conan remove...` but with an slightly different meaning 😞 
```
parser.add_argument('-p', '--packages', nargs="*", action=Extender,
                    help="Select package to remove specifying the package ID")
```

### LINKED ISSUE #5250: conan download "-p" deprecation
The same with `get` and `upload`, deprecate the `-p` in favor of full ref.

### PR DESCRIPTION
**Pull request by DodoMomo**
Changelog: Deprecated ``conan copy|download|upload <ref> -p=ID``, use ``conan .... <pref>`` instead
Docs: conan-io/docs#1317

Fixes #5250
Closes #5225 

- [X] Refer to the issue that supports this Pull Request.
- [X] I've followed the PEP8 style guides for Python code.
- [X] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one.

### CODE REVIEW

File: conans/client/conan_api.py [hunk 1]
```diff
@@ -421,24 +421,23 @@ def export_pkg(self, conanfile_path, name, channel, source_folder=None, build_fo
             raise
 
     @api_method
-    def download(self, reference, remote_name=None, package=None, recipe=False):
-        # FIXME: The "package" parameter name is very bad, it is a list of package_ids
-        if package and recipe:
-            raise ConanException("recipe parameter cannot be used together with package")
+    def download(self, reference, remote_name=None, package_list=None, recipe=False):
```
Comment (danimtb) [inline conans/client/conan_api.py]: I think we need to unify the name here of `package_list`. I think `packages` is the name used in other api methods

File: conans/test/functional/command/copy_packages_test.py [hunk 2]
```diff
@@ -27,8 +27,8 @@ class Pkg(ConanFile):
         packages = os.listdir(pkgdir)
         self.assertEqual(len(packages), 3)
 
-        # Copy just one
-        client.run("copy Hello0/0.1@lasote/stable pepe/stable -p %s" % packages[0])
+        # Copy just one with --package argument
+        client.run("copy Hello0/0.1@lasote/stable:%s pepe/stable" % packages[0])
```
Comment (memsharded) [inline conans/test/functional/command/copy_packages_test.py]: I think I would leave one test (for each command) that keeps the old ``-p=ID`` argument, to make sure that we are not breaking existing users.

Comment (DodoMomo) [reply conans/test/functional/command/copy_packages_test.py]: OK, I will take that as an opportunity to refactor the tests a bit.

**Comment by DodoMomo** [comment]
I'm sorry i messed up with the last commit.

File: conans/client/conan_api.py [hunk 3]
```diff
@@ -421,24 +421,23 @@ def export_pkg(self, conanfile_path, name, channel, source_folder=None, build_fo
...
         ref = ConanFileReference.loads(reference)
         if check_valid_ref(ref, allow_pattern=False):
-            if package and ref.revision is None:
-                for package_id in package:
+            if packages and ref.revision is None:
+                for package_id in packages:
                     if "#" in package_id:
                         raise ConanException("It is needed to specify the recipe revision if you "
-                                             "specify a package revision")
+                                             "specify a packages revision")
```
Comment (jgsogo) [inline conans/client/conan_api.py]:
```suggestion
                                             "specify a package revision")
```

**Pull request merged**